docs/conf.py

```diff
@@ -40,6 +46,32 @@
 # This pattern also affects html_static_path and html_extra_path.
 exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
 
+# -- sphinx_c_autodoc --------------------------------------------------------
+
+c_autodoc_roots = [
+    "../hpy/devel/include/common",
+    "../hpy/devel/src",
+]
+
+
+def pre_process(app, filename, contents, *args):
+    # remove HPyAPI_RUNTIME_FUNC so that the sphinx-c-autodoc and clang
+    # find and render the API functions
+    contents[:] = [
+        re.sub(r"HPyAPI_RUNTIME_FUNC\((.*)\)", r"\1", part)
+        for part in contents
+    ]
+
+
+def setup(app):
+    app.connect("c-autodoc-pre-process", pre_process)
+
+
+if 'READTHEDOCS' in os.environ:
+    from clang import cindex
+    cindex.Config.set_library_file(
+        "/usr/lib/x86_64-linux-gnu/libclang-6.0.so.1"
```

Comment:
this looks a bit fragile, but I suppose it's the only way to have it working?

Reply:
The full story is a bit involved, but yes.

Right now readthedocs uses Ubuntu 18.04 and only includes llvm 6. sphinx-c-autodoc requires python-clang >= 7 and I've make pip install clang==10.0.1. This looks for libclang-10.so, which it can't find. However, python-clang 10.0.1 seems to work fine with the libclang-6.0.so if one points it directly to the .so like we do here.

The good news is that readthedocs is busy trialing Ubuntu 20.04 for their builds and once that becomes public, we should be able to just delete these lines (or at least it currently works without these lines on my Ubuntu 20.04 machine).

Reply:
In that case, please add a TODO comment so we know when to remove this code.

Reply:
TODO added.